src/main/java/com/splunk/ibm/mq/metricscollector/PerformanceEventQueueCollector.java

```diff
@@ -0,0 +1,155 @@
+import com.ibm.mq.constants.CMQC;
+import com.ibm.mq.constants.MQConstants;
+import com.ibm.mq.headers.pcf.PCFMessage;
+import com.splunk.ibm.mq.config.QueueManager;
+import com.splunk.ibm.mq.opentelemetry.OpenTelemetryMetricWriteHelper;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import java.io.IOException;
+import org.slf4j.Logger;
+
+public final class PerformanceEventQueueCollector implements MetricsPublisher {
+
+  private static final Logger logger =
+      ExtensionsLoggerFactory.getLogger(PerformanceEventQueueCollector.class);
+  private final OpenTelemetryMetricWriteHelper metricWriteHelper;
+  private final QueueManager queueManager;
+  private final MQQueueManager mqQueueManager;
+  private final LongCounter fullQueueDepthCounter;
+  private final LongCounter highQueueDepthCounter;
+  private final LongCounter lowQueueDepthCounter;
+
+  public PerformanceEventQueueCollector(
+      MQQueueManager mqQueueManager,
+      QueueManager queueManager,
+      MetricWriteHelper metricWriteHelper) {
+    if (!(metricWriteHelper instanceof OpenTelemetryMetricWriteHelper)) {
+      throw new IllegalArgumentException(
+          "metricWriteHelper is not an instance of OpenTelemetryMetricWriteHelper");
+    }
+    this.mqQueueManager = mqQueueManager;
+    this.queueManager = queueManager;
+    this.metricWriteHelper = (OpenTelemetryMetricWriteHelper) metricWriteHelper;
+    this.fullQueueDepthCounter =
+        this.metricWriteHelper
+            .getMeter(queueManager.getName())
+            .counterBuilder("mq.queue.depth.full.event")
+            .setUnit("1")
+            .build();
+    this.highQueueDepthCounter =
+        this.metricWriteHelper
+            .getMeter(queueManager.getName())
+            .counterBuilder("mq.queue.depth.high.event")
+            .setUnit("1")
+            .build();
+    this.lowQueueDepthCounter =
+        this.metricWriteHelper
+            .getMeter(queueManager.getName())
+            .counterBuilder("mq.queue.depth.low.event")
+            .setUnit("1")
+            .build();
+  }
+
+  private void readEvents(String performanceEventsQueueName) throws Exception {
+
+    MQQueue queue = null;
+    try {
+      int queueAccessOptions = MQConstants.MQOO_FAIL_IF_QUIESCING | MQConstants.MQOO_INPUT_SHARED;
+      queue = mqQueueManager.accessQueue(performanceEventsQueueName, queueAccessOptions);
+      // keep going until receiving the exception MQConstants.MQRC_NO_MSG_AVAILABLE
+      while (true) {
+        try {
+          MQGetMessageOptions getOptions = new MQGetMessageOptions();
+          getOptions.options = MQConstants.MQGMO_NO_WAIT | MQConstants.MQGMO_FAIL_IF_QUIESCING;
+          MQMessage message = new MQMessage();
+
+          queue.get(message, getOptions);
+          PCFMessage received = new PCFMessage(message);
+          switch (received.getReason()) {
+            case CMQC.MQRC_Q_FULL:
```

Comment:
now this is just a nitpick.   feel free to do it or not.

I'm just thinking i would refactor the switch statement to a method - 
   incrementCounterByEventType(receivedMsg);